File: BAEKJOON/all_problems/17K/17264.cpp
#include <bits/stdc++.h>
using namespace std;

void solve(){
    int n, p, w, l, g;
    cin >> n >> p >> w >> l >> g;

    set<string> can_win_names;
    while(p--){
        string name, wl; cin >> name >> wl;
        if(wl == "W") can_win_names.insert(name);
    }

    int score = 0;
    for(int i=0; i<n; ++i){
        string name; cin >> name;
        if(can_win_names.count(name))   score += w;
        else                            score = max(0, score-l);

        if(score >= g){
            cout << "I AM NOT IRONMAN!!";
            return;
        }
    }

    cout << "I AM IRONMAN!!";
}

int main(void){
    ios::sync_with_stdio(0);
    cin.tie(0);
    solve();
    return 0;
}

File: BAEKJOON/all_problems/17K/17264.py
import sys
input = sys.stdin.readline

def solve():
    n, p = map(int, input().split())
    w, l, g = map(int, input().split())

    players_data = dict()
    for _ in range(p):
        name, win_or_lose = map(str, input().split())
        players_data[name] = w if win_or_lose == 'W' else -l
    # print(players_data)

    score = 0
    for _ in range(n):
        name = input().rstrip()
        if(name in players_data):   score = max(0, score + players_data[name])
        else:                       score = max(0, score - l)

        if score >= g:
            print("I AM NOT IRONMAN!!")
            return

    print("I AM IRONMAN!!")

solve()